diff --git a/pulpcore/app/wsgi.py b/pulpcore/app/wsgi.py
index aaf55613e..53107b135 100644
--- a/pulpcore/app/wsgi.py
+++ b/pulpcore/app/wsgi.py
@@ -25,15 +25,15 @@ if not using_pulp_api_worker.get(False):
 
 
 class WorkerNameMetricsExporter(OTLPMetricExporter):
-    def export(self, metrics_data, timeout_millis=10_000, **kwargs):
+    def export(self, metrics_data, **kwargs):
         for resource_metric in metrics_data.resource_metrics:
             for scope_metric in resource_metric.scope_metrics:
                 for metric in scope_metric.metrics:
-                    if metric.data.data_points:
-                        point = metric.data.data_points[0]
-                        point.attributes["worker.name"] = get_worker_name()
+                    if metric.data.data_points and isinstance(metric.data.data_points, list):
+                        for point in metric.data.data_points:
+                            point.attributes["worker.name"] = get_worker_name()
 
-        return super().export(metrics_data, timeout_millis, **kwargs)
+        return super().export(metrics_data, **kwargs)
 
 
 exporter = WorkerNameMetricsExporter()

