From c9663d0106eedd9e79e755404fcb5a9e058771f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Thu, 31 Oct 2024 11:59:46 -0300
Subject: [PATCH] Change how we filter for unblocked tasks.

---
 pulpcore/tasking/worker.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
index e77c9d2bb..f756e248e 100644
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -480,7 +480,7 @@ class PulpcoreWorker:
             ):
                 # For performance reasons we aggregate these statistics on a single database call.
                 unblocked_tasks_stats = (
-                    Task.objects.filter(unblocked_at__isnull=False, started_at__isnull=True)
+                    Task.objects.filter(unblocked_at__isnull=False, state=TASK_STATES.WAITING)
                     .annotate(unblocked_for=Value(timezone.now()) - F("unblocked_at"))
                     .aggregate(
                         longest_unblocked_waiting_time=Max(
-- 
2.47.0

