From 4a85e84d2235a5095d5edc42a3dcf297d60230e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Wed, 3 Jul 2024 10:23:16 -0300
Subject: [PATCH] Add RHServiceAccountCertAuthentication backend

---
 pulpcore/app/authentication.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/pulpcore/app/authentication.py b/pulpcore/app/authentication.py
index 93b0be8ce..c14ca7863 100644
--- a/pulpcore/app/authentication.py
+++ b/pulpcore/app/authentication.py
@@ -72,3 +72,9 @@ class JSONHeaderRemoteAuthentication(BaseAuthentication):
 
         _logger.debug(_("User {user} authenticated.").format(user=remote_user))
         return (user, None)
+
+
+class RHServiceAccountCertAuthentication(JSONHeaderRemoteAuthentication):
+
+    header = "HTTP_X_RH_IDENTITY"
+    jq_filter = ".identity.x509.subject_dn"
-- 
2.45.2

