From 5adb4d2d10f09caff1c36ab27e1f8eadae28ff94 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Fri, 25 Oct 2024 16:06:20 -0300
Subject: [PATCH] Add a log to the task metrics call.

---
 pulpcore/tasking/worker.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
index e77c9d2bb..45a8b4632 100644
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -506,6 +506,10 @@ class PulpcoreWorker:
                     unblocked_tasks_stats["longest_unblocked_waiting_time"].seconds
                 )
 
+                _logger.info(
+                    f"We have {unblocked_tasks_stats['unblocked_tasks_count_gte_threshold']} unblocked tasks and the longest waiting time for a task is {unblocked_tasks_stats['longest_unblocked_waiting_time'].seconds} seconds."
+                )
+
                 self.cursor.execute(f"NOTIFY pulp_worker_metrics_heartbeat, '{str(now)}'")
 
     def run(self, burst=False):
-- 
2.47.0

