From 442649f369f0bd693302bce73b373632b06869ab Mon Sep 17 00:00:00 2001
From: git-hyagi <45576767+git-hyagi@users.noreply.github.com>
Date: Tue, 28 Jan 2025 16:17:36 -0300
Subject: [PATCH] Add Content Sources periodic telemetry task

---
 pulpcore/tasking/_util.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/pulpcore/tasking/_util.py b/pulpcore/tasking/_util.py
index f90683415..1a6102d66 100644
--- a/pulpcore/tasking/_util.py
+++ b/pulpcore/tasking/_util.py
@@ -30,6 +30,10 @@ if settings.WORKER_TYPE == "redis":
     from pulpcore.tasking.redis_tasks import execute_task
 else:
     from pulpcore.tasking.tasks import execute_task
+from pulp_service.app.tasks.util import (
+    content_sources_periodic_telemetry,
+    rhel_ai_repos_periodic_telemetry,
+)


 _logger = logging.getLogger(__name__)
@@ -39,6 +43,8 @@ def startup_hook():
     configure_analytics()
     configure_cleanup()
     configure_periodic_telemetry()
+    content_sources_periodic_telemetry()
+    rhel_ai_repos_periodic_telemetry()


 def delete_incomplete_resources(task):
--
2.46.2
