From 5101a3b3a49bef9e05f6c7ee0b47b039d31d710c Mon Sep 17 00:00:00 2001
From: git-hyagi <45576767+git-hyagi@users.noreply.github.com>
Date: Mon, 27 Jan 2025 14:35:49 -0300
Subject: [PATCH] Log request.META in the status endpoint

---
 pulpcore/app/views/status.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pulpcore/app/views/status.py b/pulpcore/app/views/status.py
index 70b1b8dad..83f9cb02d 100644
--- a/pulpcore/app/views/status.py
+++ b/pulpcore/app/views/status.py
@@ -61,6 +61,7 @@ class StatusView(APIView):
          * disk usage information
         """
         versions = []
+        _logger.info(f"REQUEST META: {request.META}")
         for app in pulp_plugin_configs():
             versions.append(
                 {
-- 
2.46.2

