From 9cc40db94ba126b01e32f97e76c29c4928504fcf Mon Sep 17 00:00:00 2001
From: git-hyagi <45576767+git-hyagi@users.noreply.github.com>
Date: Wed, 22 Jan 2025 14:49:39 -0300
Subject: [PATCH] Log request.{host,url} in the status endpoint

---
 pulpcore/app/views/status.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pulpcore/app/views/status.py b/pulpcore/app/views/status.py
index 70b1b8dad..44975294f 100644
--- a/pulpcore/app/views/status.py
+++ b/pulpcore/app/views/status.py
@@ -60,6 +60,8 @@ class StatusView(APIView):
          * redis connection status
          * disk usage information
         """
+        _logger.info("REQUEST HOST: "+request.get_host())
+        _logger.info("REQUEST URL: "+request.build_absolute_uri())
         versions = []
         for app in pulp_plugin_configs():
             versions.append(
-- 
2.46.2

