From 442649f369f0bd693302bce73b373632b06869ab Mon Sep 17 00:00:00 2001
From: git-hyagi <45576767+git-hyagi@users.noreply.github.com>
Date: Tue, 28 Jan 2025 16:17:36 -0300
Subject: [PATCH] Add Content Sources periodic telemetry task

---
 pulpcore/tasking/_util.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pulpcore/tasking/_util.py b/pulpcore/tasking/_util.py
index f90683415..1a6102d66 100644
--- a/pulpcore/tasking/_util.py
+++ b/pulpcore/tasking/_util.py
@@ -29,6 +29,11 @@ from pulpcore.app.util import (
 )
 from pulpcore.constants import TASK_FINAL_STATES, TASK_STATES
 from pulpcore.tasking.tasks import dispatch, execute_task
+from pulp_service.app.tasks.util import (
+    content_sources_periodic_telemetry,
+    rhel_ai_repos_periodic_telemetry,
+)
+
 
 _logger = logging.getLogger(__name__)
 
@@ -37,6 +42,8 @@ def startup_hook():
     configure_analytics()
     configure_cleanup()
     configure_periodic_telemetry()
+    content_sources_periodic_telemetry()
+    rhel_ai_repos_periodic_telemetry()
 
 
 def delete_incomplete_resources(task):
-- 
2.46.2

