From d02b717078402b6132268a9d8fba6229c5a54777 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Thu, 30 Jan 2025 16:03:08 -0300
Subject: [PATCH] Change the TASK_GRACE_INTERVAL to 360 heartbeats, approx. 1h.

---
 pulpcore/tasking/worker.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
index bbf70458a..a7a4f500e 100644
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -46,7 +46,7 @@ random.seed()
 # they will live as constants instead of "proper" settings.
 
 # Number of heartbeats for a task to finish on graceful worker shutdown (approx)
-TASK_GRACE_INTERVAL = 3
+TASK_GRACE_INTERVAL = 360
 # Number of heartbeats between attempts to kill the subprocess (approx)
 TASK_KILL_INTERVAL = 1
 # Number of heartbeats between cleaning up worker processes (approx)
@@ -384,7 +384,7 @@ class PulpcoreWorker:
                     else:
                         self.task_grace_timeout = TASK_KILL_INTERVAL
                         _logger.info(
-                            "Aborting current task %s in domain: %s due to cancelation.",
+                            "Aborting current task %s in domain: %s due to cancellation.",
                             task.pk,
                             domain.name,
                         )
-- 
2.48.1

