From 6835b7d0d1a3f1f1961b69dafd138bfe01a9e9bb Mon Sep 17 00:00:00 2001
From: git-hyagi <45576767+git-hyagi@users.noreply.github.com>
Date: Mon, 24 Feb 2025 16:17:38 -0300
Subject: [PATCH] Fix pulp_content_origin_with_prefix fixture

---
 pulpcore/pytest_plugin.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/pulpcore/pytest_plugin.py b/pulpcore/pytest_plugin.py
index 06875297f..fd72f2017 100644
--- a/pulpcore/pytest_plugin.py
+++ b/pulpcore/pytest_plugin.py
@@ -750,12 +750,14 @@ def pulp_domain_enabled(pulp_settings):
 
 
 @pytest.fixture(scope="session")
-def pulp_content_origin(pulp_settings):
-    return pulp_settings.CONTENT_ORIGIN
+def pulp_content_origin(pulp_settings, bindings_cfg):
+    return pulp_settings.CONTENT_ORIGIN if pulp_settings.CONTENT_ORIGIN else bindings_cfg.host
 
 
 @pytest.fixture(scope="session")
-def pulp_content_origin_with_prefix(pulp_settings):
+def pulp_content_origin_with_prefix(pulp_settings, bindings_cfg):
+    if not pulp_settings.CONTENT_ORIGIN:
+        return bindings_cfg.host + pulp_settings.CONTENT_PATH_PREFIX[:-1]
     return pulp_settings.CONTENT_ORIGIN + pulp_settings.CONTENT_PATH_PREFIX[:-1]
 
 
-- 
2.46.2

