diff --git a/pulp_container/app/token_verification.py b/pulp_container/app/token_verification.py
index 019310ac..9c685b68 100644
--- a/pulp_container/app/token_verification.py
+++ b/pulp_container/app/token_verification.py
@@ -14,6 +14,7 @@ from rest_framework.exceptions import AuthenticationFailed, NotAuthenticated
 from rest_framework.permissions import SAFE_METHODS, BasePermission
 
 from pulp_container.app.utils import get_full_path
+from pulp_service.app.authorization import DomainBasedPermission
 
 Scope = namedtuple("Scope", "resource_type, name, action")
 User = get_user_model()
@@ -177,23 +178,11 @@ def get_scopes(request):
     return scopes
 
 
-class RegistryPermission(BasePermission):
+class RegistryPermission(DomainBasedPermission):
     """
     Permission class to determine permissions based on the request user.
     """
 
-    message = "Access to the requested resource is not authorized."
-
-    def has_permission(self, request, view):
-        """
-        Decide upon permission based on user.
-        """
-        if request.user.is_superuser:
-            return True
-        if request.method in SAFE_METHODS:
-            return True
-
-        return False
 
 
 class TokenPermission(BasePermission):
