From 57bd42334175865c623b109453114a1e86fda338 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Fri, 13 Jun 2025 10:57:03 -0300
Subject: [PATCH] Log the execution plan for the Task select query.

---
 pulpcore/tasking/worker.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
index dd61cd6ff..1612fb37d 100644
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -324,10 +324,14 @@ class PulpcoreWorker:
         """Iterate over ready tasks and yield each task while holding the lock."""
         while not self.shutdown_requested:
             # When batching this query, be sure to use "pulp_created" as a cursor
-            for task in Task.objects.filter(
+            task_query = Task.objects.filter(
                 state__in=TASK_INCOMPLETE_STATES,
                 unblocked_at__isnull=False,
-            ).order_by("-immediate", "pulp_created"):
+            ).order_by("-immediate", "pulp_created")
+ 
+            _logger.info(task_query.explain()) # We want to check the execution plan for this query
+            
+            for task in task_query:
                 # This code will only be called if we acquired the lock successfully
                 # The lock will be automatically be released at the end of the block
                 with contextlib.suppress(AdvisoryLockError), task:
-- 
2.49.0

