From d2578a51b604d478fae11c20fb7bb8da022bf7ce Mon Sep 17 00:00:00 2001
From: Yasen <yasen@trahnov.eu>
Date: Fri, 8 Aug 2025 15:44:15 +0200
Subject: [PATCH] Add auth override to pulp-python SimpleView

---
 pulp_python/app/pypi/views.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pulp_python/app/pypi/views.py b/pulp_python/app/pypi/views.py
index 79033d4..0e8a747 100644
--- a/pulp_python/app/pypi/views.py
+++ b/pulp_python/app/pypi/views.py
@@ -220,6 +220,8 @@ class SimpleView(PackageUploadMixin, ViewSet):
     """View for the PyPI simple API."""
 
     endpoint_name = "simple"
+    authentication_classes = []
+    permission_classes = []
     DEFAULT_ACCESS_POLICY = {
         "statements": [
             {
-- 
2.50.1