From 06cac5ec9641093058ac3a833751d6e5c70b9916 Mon Sep 17 00:00:00 2001
From: Dennis Kliban <dkliban@redhat.com>
Date: Mon, 21 Jul 2025 07:59:20 -0400
Subject: [PATCH] Added call to self.beat() in handle_available_tasks().

---
 pulpcore/tasking/worker.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
index dd61cd6ff..d60209290 100644
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -483,6 +483,7 @@ class PulpcoreWorker:
         """
         keep_looping = True
         while keep_looping and not self.shutdown_requested:
+            self.beat()
             try:
                 with PGAdvisoryLock(TASK_UNBLOCKING_LOCK):
                     keep_looping = self.unblock_tasks()
-- 
2.50.0

