From 20ee3e3c769ae070ec2711b76e5797367f230ed6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Fri, 13 Jun 2025 12:56:04 -0300
Subject: [PATCH 2/2] Try a different approach to get a task and lock it.

---
 pulpcore/tasking/worker.py | 82 +++++++++++++++++++++-----------------
 1 file changed, 45 insertions(+), 37 deletions(-)

diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
index 1612fb37d..442b93717 100644
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -13,7 +13,7 @@ from tempfile import TemporaryDirectory
 from packaging.version import parse as parse_version
 
 from django.conf import settings
-from django.db import connection
+from django.db import connection, transaction
 from django.db.models import Case, Count, F, Max, Value, When
 from django.utils import timezone
 
@@ -324,47 +324,55 @@ class PulpcoreWorker:
         """Iterate over ready tasks and yield each task while holding the lock."""
         while not self.shutdown_requested:
             # When batching this query, be sure to use "pulp_created" as a cursor
-            task_query = Task.objects.filter(
-                state__in=TASK_INCOMPLETE_STATES,
+            task_query = Task.objects.select_for_update(skip_locked=True).filter(
+                state__in=TASK_STATES.WAITING,
                 unblocked_at__isnull=False,
-            ).order_by("-immediate", "pulp_created")
+                worker__isnull=True,
+            ).order_by("-immediate", "pulp_created")[0]
  
             _logger.info(task_query.explain()) # We want to check the execution plan for this query
             
-            for task in task_query:
-                # This code will only be called if we acquired the lock successfully
-                # The lock will be automatically be released at the end of the block
-                with contextlib.suppress(AdvisoryLockError), task:
-                    # Check if someone else changed the task before we got the lock
-                    task.refresh_from_db()
-
-                    if task.state == TASK_STATES.CANCELING and task.worker is None:
-                        # No worker picked this task up before being canceled
-                        if self.cancel_abandoned_task(task, TASK_STATES.CANCELED):
-                            # Continue looking for the next task without considering this
-                            # tasks resources, as we just released them
-                            continue
-                    if task.state in [TASK_STATES.RUNNING, TASK_STATES.CANCELING]:
-                        # A running task without a lock must be abandoned
-                        if self.cancel_abandoned_task(
-                            task, TASK_STATES.FAILED, "Worker has gone missing."
+            with transaction.atomic():
+                for task in task_query:
+
+                    self.cancel_task = False
+                    self.task = task
+                    task.worker = self.worker
+                    task.save(update_fields=["worker"])
+                    
+                    # This code will only be called if we acquired the lock successfully
+                    # The lock will be automatically be released at the end of the block
+                    with contextlib.suppress(AdvisoryLockError), task:
+                        # Check if someone else changed the task before we got the lock
+                        # task.refresh_from_db()
+
+                        # if task.state == TASK_STATES.CANCELING and task.worker is None:
+                        #     # No worker picked this task up before being canceled
+                        #     if self.cancel_abandoned_task(task, TASK_STATES.CANCELED):
+                        #         # Continue looking for the next task without considering this
+                        #         # tasks resources, as we just released them
+                        #         continue
+                        # if task.state in [TASK_STATES.RUNNING, TASK_STATES.CANCELING]:
+                        #     # A running task without a lock must be abandoned
+                        #     if self.cancel_abandoned_task(
+                        #         task, TASK_STATES.FAILED, "Worker has gone missing."
+                        #     ):
+                        #         # Continue looking for the next task without considering this
+                        #         # tasks resources, as we just released them
+                        #         continue
+
+                        # This statement is using lazy evaluation
+                        if (
+                            task.state == TASK_STATES.WAITING
+                            and task.unblocked_at is not None
+                            and self.is_compatible(task)
                         ):
-                            # Continue looking for the next task without considering this
-                            # tasks resources, as we just released them
-                            continue
-
-                    # This statement is using lazy evaluation
-                    if (
-                        task.state == TASK_STATES.WAITING
-                        and task.unblocked_at is not None
-                        and self.is_compatible(task)
-                    ):
-                        yield task
-                        # Start from the top of the Task list
-                        break
-            else:
-                # No task found in the for-loop
-                break
+                            yield task
+                            # Start from the top of the Task list
+                            break
+                else:
+                    # No task found in the for-loop
+                    break
 
     def sleep(self):
         """Wait for signals on the wakeup channel while heart beating."""
-- 
2.49.0

