From e9c6c866148569aad8349dadc79cde39faf5a0c7 Mon Sep 17 00:00:00 2001
From: git-hyagi <45576767+git-hyagi@users.noreply.github.com>
Date: Mon, 22 Sep 2025 12:42:32 -0300
Subject: [PATCH] Print content request.headers if basic auth request

---
 pulpcore/content/authentication.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pulpcore/content/authentication.py b/pulpcore/content/authentication.py
index 016c2990c..88dbb5dff 100644
--- a/pulpcore/content/authentication.py
+++ b/pulpcore/content/authentication.py
@@ -39,6 +39,8 @@ async def authenticate(request, handler):
             try:
                 domain = validate_domain(request)
                 fake_view.perform_authentication(drf_request)
+                if 'Authorization' in request.headers:
+                    log.info("####### Authorization header present!")
             except (InterfaceError, DatabaseError):
                 Handler._reset_db_connection()
                 domain = validate_domain(request)
-- 
2.46.2

