diff --git a/pulp_python/app/pypi/views.py b/pulp_python/app/pypi/views.py
index 4bbbb8a6..929bfece 100644
--- a/pulp_python/app/pypi/views.py
+++ b/pulp_python/app/pypi/views.py
@@ -57,8 +57,7 @@ from pulp_python.app import tasks
 
 log = logging.getLogger(__name__)
 
-ORIGIN_HOST = settings.CONTENT_ORIGIN if settings.CONTENT_ORIGIN else settings.PYPI_API_HOSTNAME
-BASE_CONTENT_URL = urljoin(ORIGIN_HOST, settings.CONTENT_PATH_PREFIX)
+BASE_CONTENT_URL = settings.CONTENT_PATH_PREFIX
 BASE_API_URL = urljoin(settings.PYPI_API_HOSTNAME, "pypi/")
 
 PYPI_SIMPLE_V1_HTML = "application/vnd.pypi.simple.v1+html"