diff --git a/pulp_python/app/pypi/views.py b/pulp_python/app/pypi/views.py
index b7808a9..531d594 100644
--- a/pulp_python/app/pypi/views.py
+++ b/pulp_python/app/pypi/views.py
@@ -54,6 +54,7 @@ from pulp_python.app.utils import (
 )
 
 from pulp_python.app import tasks
+from pulp_service.app.authorization import AllowUnauthPull,DomainBasedPermission
 
 log = logging.getLogger(__name__)
 
@@ -78,6 +79,8 @@ class PyPIMixin:
 
     _distro = None
 
+    permission_classes = [AllowUnauthPull|DomainBasedPermission]
+
     @property
     def distribution(self):
         if self._distro:
