diff --git a/pulpcore/app/entrypoint.py b/pulpcore/app/entrypoint.py
--- a/pulpcore/app/entrypoint.py
+++ b/pulpcore/app/entrypoint.py
@@ -29,12 +29,12 @@
     def heartbeat(self):
         try:
             self.app_status.save_heartbeat()
-            logger.debug(self.beat_msg)
+            logger.info(self.beat_msg)
         except (InterfaceError, DatabaseError):
             connection.close_if_unusable_or_obsolete()
             try:
                 self.app_status.save_heartbeat()
-                logger.debug(self.beat_msg)
+                logger.info(self.beat_msg)
             except (InterfaceError, DatabaseError) as e:
                 logger.error(f"{self.fail_beat_msg} Exception: {str(e)}")
                 exit(Arbiter.WORKER_BOOT_ERROR)
diff --git a/pulpcore/content/__init__.py b/pulpcore/content/__init__.py
--- a/pulpcore/content/__init__.py
+++ b/pulpcore/content/__init__.py
@@ -71,12 +71,12 @@
             await asyncio.sleep(heartbeat_interval)
             try:
                 await app_status.asave_heartbeat()
-                log.debug(msg)
+                log.info(msg)
             except (InterfaceError, DatabaseError):
                 await sync_to_async(Handler._reset_db_connection)()
                 try:
                     await app_status.asave_heartbeat()
-                    log.debug(msg)
+                    log.info(msg)
                 except (InterfaceError, DatabaseError) as e:
                     log.error(f"{fail_msg} Exception: {str(e)}")
                     exit(Arbiter.WORKER_BOOT_ERROR)
diff --git a/pulpcore/tasking/redis_worker.py b/pulpcore/tasking/redis_worker.py
--- a/pulpcore/tasking/redis_worker.py
+++ b/pulpcore/tasking/redis_worker.py
@@ -201,7 +201,7 @@
         )
         try:
             self.app_status.save_heartbeat()
-            _logger.debug(msg)
+            _logger.info(msg)
         except (IntegrityError, DatabaseError):
             _logger.error(f"Updating the heartbeat of worker {self.name} failed.")
             self.shutdown_requested = True
diff --git a/pulpcore/tasking/worker.py b/pulpcore/tasking/worker.py
--- a/pulpcore/tasking/worker.py
+++ b/pulpcore/tasking/worker.py
@@ -194,7 +194,7 @@
         )
         try:
             self.app_status.save_heartbeat()
-            _logger.debug(msg)
+            _logger.info(msg)
         except (IntegrityError, DatabaseError):
             # WARNING: Do not attempt to recycle the connection here.
             # The advisory locks are bound to the connection and we must not loose them.
