diff --git a/pulp_rpm/app/viewsets/repository.py b/pulp_rpm/app/viewsets/repository.py
index 335e518fa..60a557f58 100644
--- a/pulp_rpm/app/viewsets/repository.py
+++ b/pulp_rpm/app/viewsets/repository.py
@@ -56,7 +56,7 @@ class RpmModifyRepositoryActionMixin(ModifyRepositoryActionMixin):
     @action(detail=True, methods=["post"], serializer_class=RepositoryAddRemoveContentSerializer)
     def modify(self, request, pk):
         add_content_units = request.data.get("add_content_units", [])
-        package_ids = [extract_pk(href) for href in add_content_units if "/packages/" in href]
+        package_ids = [self._uri_parse(href) for href in add_content_units if "/packages/" in href]
         repository = self.get_object()
 
         if package_ids and repository.package_signing_service:
@@ -70,6 +70,11 @@ class RpmModifyRepositoryActionMixin(ModifyRepositoryActionMixin):
 
         return super().modify(request, pk)
 
+    def _uri_parse(self, uri):
+        if uri.startswith("prn:"):
+            return extract_pk(uri)
+        return uri[-37:-1]
+
 
 class RpmRepositoryViewSet(RepositoryViewSet, RpmModifyRepositoryActionMixin, RolesMixin):
     """
