diff --git a/pulp_python/app/models.py b/pulp_python/app/models.py
index fa45835..9923437 100644
--- a/pulp_python/app/models.py
+++ b/pulp_python/app/models.py
@@ -391,10 +391,21 @@ class PythonRepository(Repository, AutoAddObjPermsMixin):
 
         # avoid circular import issues
         from pulp_python.app import tasks
+        from pulpcore.plugin.tasking import dispatch
+        from pulpcore.plugin.models import CreatedResource
 
         if self.autopublish:
             tasks.publish(repository_version_pk=version.pk)
 
+        if self.pulp_labels.get("agent_scan") == "true":
+            for content in version.added():
+                task = dispatch(
+                    tasks.scan_package,
+                    exclusive_resources=[self],
+                    kwargs={"repo_version_pk": version.pk, "content_pk": str(content.pk)},
+                )
+                CreatedResource.objects.create(content_object=task)
+
     def finalize_new_version(self, new_version):
         """
         Remove duplicate packages that have the same filename.
diff --git a/pulp_python/app/tasks/__init__.py b/pulp_python/app/tasks/__init__.py
index 6c2949e..bcae67e 100644
--- a/pulp_python/app/tasks/__init__.py
+++ b/pulp_python/app/tasks/__init__.py
@@ -7,3 +7,4 @@ from .repair import repair  # noqa:F401
 from .sync import sync  # noqa:F401
 from .upload import upload, upload_group  # noqa:F401
 from .vulnerability_report import get_repo_version_content  # noqa:F401
+from .scan_package import scan_package # noqa:F401
