From ed3492914f4095698ee6762257f293fb546c3fb2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20=22decko=22=20de=20Brito?= <decko@redhat.com>
Date: Wed, 6 Mar 2024 15:35:30 -0300
Subject: [PATCH] Enable logging header info

---
 pulpcore/app/authentication.py | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/pulpcore/app/authentication.py b/pulpcore/app/authentication.py
index 93b0be8ce..5d53f5252 100644
--- a/pulpcore/app/authentication.py
+++ b/pulpcore/app/authentication.py
@@ -40,7 +40,7 @@ class JSONHeaderRemoteAuthentication(BaseAuthentication):
         to be filtered by JQ.
         """
         if self.header not in request.META:
-            _logger.debug(
+            _logger.info(
                 "Access not allowed. Header {header} not found.".format(header=self.header)
             )
             return None
@@ -48,8 +48,10 @@ class JSONHeaderRemoteAuthentication(BaseAuthentication):
         try:
             header_content = request.META.get(self.header)
             header_decoded_content = b64decode(header_content)
+            _logger.info(f"Header: {self.header}")
+            _logger.info(f"Header value: {header_decoded_content}")
         except Base64DecodeError:
-            _logger.debug(_("Access not allowed - Header content is not Base64 encoded."))
+            _logger.info(_("Access not allowed - Header content is not Base64 encoded."))
             raise AuthenticationFailed(_("Access denied."))
 
         try:
@@ -58,11 +60,11 @@ class JSONHeaderRemoteAuthentication(BaseAuthentication):
 
             remote_user = json_path.input_value(header_value).first()
         except json.JSONDecodeError:
-            _logger.debug(_("Access not allowed - Invalid JSON."))
+            _logger.info(_("Access not allowed - Invalid JSON."))
             raise AuthenticationFailed(_("Access denied. Invalid JSON."))
 
         if not remote_user:
-            _logger.debug(_("Path or value not found."))
+            _logger.info(_("Path or value not found."))
             return None
 
         user = authenticate(request, remote_user=remote_user)
@@ -70,5 +72,5 @@ class JSONHeaderRemoteAuthentication(BaseAuthentication):
         if not user:
             return None
 
-        _logger.debug(_("User {user} authenticated.").format(user=remote_user))
+        _logger.info(_("User {user} authenticated.").format(user=remote_user))
         return (user, None)
-- 
2.44.0

