diff --git a/CHANGES/4881.misc b/CHANGES/4881.misc
diff --git a/pulpcore/tests/functional/__init__.py b/pulpcore/tests/functional/__init__.py
index 214f1e7fc5..ac5360e75e 100644
--- a/pulpcore/tests/functional/__init__.py
+++ b/pulpcore/tests/functional/__init__.py
@@ -919,6 +919,11 @@ def pulp_domain_enabled(pulp_settings):
     return pulp_settings.DOMAIN_ENABLED
 
 
+@pytest.fixture(scope="session")
+def pulp_content_origin(pulp_settings):
+    return pulp_settings.CONTENT_ORIGIN
+
+
 @pytest.fixture(scope="session")
 def pulp_api_v3_path(pulp_settings, pulp_domain_enabled):
     if pulp_domain_enabled:
@@ -1072,7 +1077,7 @@ def _add_to_filesystem_cleanup(path):
 
 
 @pytest.fixture(scope="session")
-def download_content_unit(bindings_cfg, pulp_domain_enabled):
+def download_content_unit(pulp_domain_enabled, pulp_content_origin):
     def _download_content_unit(base_path, content_path, domain="default"):
         async def _get_response(url):
             async with aiohttp.ClientSession() as session:
@@ -1081,7 +1086,7 @@ async def _get_response(url):
 
         if pulp_domain_enabled:
             url_fragments = [
-                bindings_cfg.host,
+                pulp_content_origin,
                 "pulp/content",
                 domain,
                 base_path,
@@ -1089,7 +1094,7 @@ async def _get_response(url):
             ]
         else:
             url_fragments = [
-                bindings_cfg.host,
+                pulp_content_origin,
                 "pulp/content",
                 base_path,
                 content_path,
