diff --git a/pulpcore/app/urls.py b/pulpcore/app/urls.py
index 8e1138be28..d628c51890 100644
--- a/pulpcore/app/urls.py
+++ b/pulpcore/app/urls.py
@@ -13,7 +13,13 @@
 from rest_framework.routers import APIRootView
 
 from pulpcore.app.apps import pulp_plugin_configs
-from pulpcore.app.views import OrphansView, PulpImporterImportCheckView, RepairView, StatusView
+from pulpcore.app.views import (
+    LivezView,
+    OrphansView,
+    PulpImporterImportCheckView,
+    RepairView,
+    StatusView,
+)
 from pulpcore.app.viewsets import (
     ListRepositoryVersionViewSet,
     OrphansCleanupViewset,
@@ -157,6 +163,7 @@ class PulpDefaultRouter(routers.DefaultRouter):
 ]
 
 docs_and_status = [
+    path("livez/", LivezView.as_view()),
     path("status/", StatusView.as_view()),
     path(
         "docs/api.json",
diff --git a/pulpcore/app/views/__init__.py b/pulpcore/app/views/__init__.py
index d21700007a..d0357f42a9 100644
--- a/pulpcore/app/views/__init__.py
+++ b/pulpcore/app/views/__init__.py
@@ -1,4 +1,4 @@
 from .orphans import OrphansView
-from .status import StatusView
+from .status import LivezView, StatusView
 from .repair import RepairView
 from .importer import PulpImporterImportCheckView
diff --git a/pulpcore/app/views/status.py b/pulpcore/app/views/status.py
index 0b63717aa5..70b1b8dadd 100644
--- a/pulpcore/app/views/status.py
+++ b/pulpcore/app/views/status.py
@@ -136,3 +136,24 @@ def _get_redis_conn_status():
             return False
         else:
             return True
+
+
+class LivezView(APIView):
+    """
+    Liveness Probe for the REST API.
+    """
+
+    # allow anyone to access the liveness api
+    authentication_classes = []
+    permission_classes = []
+
+    @extend_schema(
+        summary="Inspect liveness of Pulp's REST API.",
+        operation_id="livez_read",
+        responses={200: None},
+    )
+    def get(self, request):
+        """
+        Returns 200 OK when API is alive.
+        """
+        return Response()
diff --git a/pulpcore/tests/functional/api/test_status.py b/pulpcore/tests/functional/api/test_status.py
index 624ae4c806..e55997ac5e 100644
--- a/pulpcore/tests/functional/api/test_status.py
+++ b/pulpcore/tests/functional/api/test_status.py
@@ -175,3 +175,15 @@ def verify_get_response(status, expected_schema):
     else:
         assert status["storage"]["free"] is not None
         assert status["storage"]["total"] is not None
+
+
+@pytest.mark.parallel
+def test_livez_unauthenticated(
+    pulpcore_bindings,
+    anonymous_user,
+):
+    """
+    Assert that GET requests to Livez API return 200 without a response body.
+    """
+    with anonymous_user:
+        assert pulpcore_bindings.LivezApi.livez_read() is None
