diff --git a/pulpcore/app/global_access_conditions.py b/pulpcore/app/global_access_conditions.py
index 3edf264292..5720f78eb0 100644
--- a/pulpcore/app/global_access_conditions.py
+++ b/pulpcore/app/global_access_conditions.py
@@ -426,7 +426,7 @@ def has_repo_or_repo_ver_param_model_or_domain_or_obj_perms(request, view, actio
     if repository := serializer.validated_data.get("repository"):
         return request.user.has_perm(permission, repository)
     elif repo_ver := serializer.validated_data.get("repository_version"):
-        return request.user.has_perm(permission, repo_ver.repository)
+        return request.user.has_perm(permission, repo_ver.repository.cast())
     return True
