diff --git a/pulpcore/app/views/status.py b/pulpcore/app/views/status.py
index 0b63717aa5..b800b38d23 100644
--- a/pulpcore/app/views/status.py
+++ b/pulpcore/app/views/status.py
@@ -43,6 +43,12 @@ class StatusView(APIView):
     authentication_classes = []
     permission_classes = []

+    def head(self, request):
+        """
+        Lightweight status check that ensures that the REST API can connect to the database.
+        """
+        return Response()
+
     @extend_schema(
         summary="Inspect status of Pulp",
         operation_id="status_read",
